crates/astria-conductor/src/block_cache.rs

```diff
@@ -0,0 +1,220 @@
+//! A cache of sequencer blocks that are only yielded in sequential order.
```

Comment:
is this something that can be re-used for celestia blocks?

Reply:
yeah